frontend/src/metabase/ui/components/theme/ColorSchemeProvider/ColorSchemeProvider.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  type ReactNode,
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import { useMedia } from "react-use";
+import { noop } from "underscore";
+
+import { isEmbeddingSdk } from "metabase/embedding-sdk/config";
+import { getIsEmbeddingIframe } from "metabase/selectors/embed";
+
+export type ResolvedColorScheme = "light" | "dark";
+
+export type ColorScheme = "auto" | ResolvedColorScheme;
+
+interface ColorSchemeContextType {
+  colorScheme: ColorScheme;
+  resolvedColorScheme: ResolvedColorScheme;
+  systemColorScheme: ResolvedColorScheme;
+  setColorScheme: (scheme: ColorScheme) => void;
+  toggleColorScheme: () => void;
+}
+
+const defaultValue: ColorSchemeContextType = {
+  colorScheme: "light",
+  resolvedColorScheme: "light",
+  systemColorScheme: "light",
+  setColorScheme: noop,
+  toggleColorScheme: noop,
+};
+
+const ColorSchemeContext = createContext<ColorSchemeContextType>(defaultValue);
+
+interface ColorSchemeProviderProps {
+  children: ReactNode;
+  defaultColorScheme?: ColorScheme;
+  forceColorScheme?: ResolvedColorScheme | null;
+}
+
+const getNextScheme = (scheme: ResolvedColorScheme) => {
+  switch (scheme) {
+    case "light":
+      return "dark";
+    case "dark":
+      return "light";
+  }
+};
```

Comment:
probably don't need a `switch` here?

```suggestion
const getNextScheme = (scheme: ResolvedColorScheme) => scheme === "dark" ? "light" : "dark";
```